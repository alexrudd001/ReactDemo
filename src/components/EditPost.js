import React, { useState } from 'react';
import PostItemForm from './PostItemForm';
import ShowMessage from './ShowMessage';

const EditPost = (props) => {
    const id = props.match.params.id;
    const userId = props.location.state.userId;
    const [ messageType, setMessageType ] = useState('');
    const [ loader, setLoader ] = useState(false);

    const onSubmitHandler = ({ title, body }) => {
        setMessageType('');
        setLoader(true);
        fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
            method: 'PUT',
            body: JSON.stringify({
                id: id,
                title: title,
                body: body,
                userId: userId
            }),
            headers: {
                'Content-type': 'application/json; charset=UTF-8'
            }
        })
            .then((response) => response.json())
            .then((post) => {
                setMessageType('success');
                setLoader(false);
            })
            .catch((error) => {
                setMessageType('error');
                setLoader(false);
            });
    };
    return (
        <div>
            <div className="divStyle">
                <PostItemForm
                    onSubmit={onSubmitHandler}
                    title={props.location.state.title}
                    body={props.location.state.body}
                />
                <ShowMessage messageType={messageType} />
            </div>
            {loader && <div className="loader" />}
        </div>
    );
};

export default EditPost;
