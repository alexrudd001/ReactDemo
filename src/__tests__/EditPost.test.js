import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditPost from '../components/EditPost';

test('renders EditPost correctly', () => {
    const props = { match: { params: { id: 1 } }, location: { state: { userId: 1 } } };
    const wrapper = shallow(<EditPost {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

