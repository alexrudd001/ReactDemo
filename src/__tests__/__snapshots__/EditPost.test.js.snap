// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditPost correctly 1`] = `
<div>
  <div
    className="divStyle"
  >
    <PostItemForm
      onSubmit={[Function]}
    />
    <ShowMessage
      messageType=""
    />
  </div>
</div>
`;
